src/shared.ts

```diff
@@ -644,3 +645,38 @@ export async function BlockBlueVerified(user: BlueBlockerUser, config: Config) {
 		}
 	}
 }
+
+// Add support for OldTwitter requests.
+window.addEventListener('message', function (ev) {
+	if (ev.data.type !== 'OLDTWITTER_REQUEST_LOAD') return;
+	if (!ev.data.url || !ev.data.body || !ev.data.headers)
+		return console.error(logstr, 'OldTwitter sent an invalid payload.', ev.data);
+
+	const body_str = JSON.stringify(ev.data.body);
+
+	api.storage.sync
+		.get(DefaultOptions)
+		.then((_config) => {
+			const config = _config as Config;
+
+			document.dispatchEvent(
+				new CustomEvent('blue-blocker-event', {
+					detail: {
+						parsedUrl: /(.+)/.exec(ev.data.url)!, // Have to turn the endpoint string into a regex result...
+						url: ev.data.url,
+						body: body_str as XMLHttpRequest['response'],
+						request: {
+							headers: ev.data.headers,
+						},
+						// OldTwitter only emits messages on success.
+						status: 200,
+					},
+				}),
+			);
+
+			ParseTimelineTweet(ev.data, config);
+		})
+		.catch((err) => {
+			console.error(logstr, 'unexpected error occurred while processing OldTwitter request:', err);
```

Comment:
Might be worthwhile to add:
```typescript
api.storage.local.set({
    [EventKey]: {
        type: ErrorEvent,
	},
});
```
here, so that the user gets a toast notification if something goes wrong. I don't think it's useful near the other error state, because that deals with OldTwitter malforming it's own data.

Reply:
Realising this entire block isn't needed since the config was only required for that ParseTimelineTweet function. This simplifies the code a lot